src/raft_server.c

```diff
@@ -1277,6 +1277,21 @@ int raft_begin_snapshot(raft_server_t *me_)
     return 0;
 }
 
+int raft_cancel_snapshot(raft_server_t *me_)
+{
+    raft_server_private_t* me = (raft_server_private_t*)me_;
+
+    if (!me->snapshot_in_progress)
+        return -1;
+
+    me->snapshot_last_idx = me->saved_snapshot_last_idx;
+    me->snapshot_last_term = me->saved_snapshot_last_term;
+
+    me->snapshot_in_progress = 1;
```

Comment:
shouldn't this be 0?

I was thinking it should work like this:
```c
raft_begin_snapshot(raft);
assert(raft_snapshot_is_in_progress(raft) == 1);
raft_cancel_snapshot(raft);
assert(raft_snapshot_is_in_progress(raft) == 0);
```

Reply:
my bad, failed commit. Also added a missing test and false assert.